src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltype/InputIllegalTypeGenerics.java

```diff
@@ -0,0 +1,59 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.illegaltype;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.function.Consumer;
+
+/**
+ * This test-input is intended to be checked using following configuration:
+ *
+ * illegalClassNames = {Boolean, Foo, Serializable}
+ * memberModifiers = {LITERAL_PUBLIC, FINAL}
+ *
+ */
+public abstract class InputIllegalTypeGenerics {
+
+    private Set<Boolean> privateSet; // OK
+    private java.util.List<Map<Boolean, Foo>> privateList; // OK
+    public Set<Boolean> set; // warn
+    public java.util.List<Map<Boolean, Foo>> list; // warn
+
+    private void methodCall() {
+        Bounded.<Boolean>foo(); // warn
+        final Consumer<Foo> consumer = Foo<Boolean>::foo; // warn
+    }
+
+    public <T extends Boolean, U extends Serializable> void typeParameter(T a) {} // warn
+
+    public void fullName(java.util.ArrayList<? super Boolean> a) {} // warn
+
+    public abstract Set<Boolean> shortName(Set<? super Boolean> a); // warn
+
+    public Set<? extends Foo> typeArgument() { // warn
+        return new TreeSet<Foo<Boolean>>(); // OK
+    }
+
+    public class MyClass<Foo extends Boolean> {} // OK
```

Comment:
Boolean creates a violation in other places, why not here?

Reply:
Indeed. From the [docs](https://checkstyle.org/config_coding.html#IllegalType):
> Checks that particular classes are never used as types in variable declarations, one-dimensional and multi-dimensional arrays, return values or parameters.

I think it is a subject to discuss. In a separate issue, maybe.

Reply:
@romani what do you think, should this check also forbid declarations like this:
```
// illegalClassNames="Boolean, Foo, Serializable"
class Foo<T extends Boolean & Serializable> {}
```

Reply:
this is corner case .....
this is declaration not real usage of this type .... we need to come to original reason of Check is it searching mostly for type instance/usage or just any possible reference .... 
lets postpone this to separate issue to let us merge this long lasting PR.

Reply:
ok, we can ignore this item then